massa-module-cache/src/hd_cache.rs

```diff
@@ -36,7 +36,7 @@ macro_rules! metadata_key {
 
 pub(crate) struct HDCache {
     /// RocksDB database
-    db: DB,
+    db: Option<DB>,
```

Comment:
why it is now Option<T> ?

Reply:
In order to destroy the DB to reset it, I needed a way to drop the current DB, which is not possible without the Option as I can't fully drop the `HDCache`.
The only alternative I thought was to iterate through the DB to remove elements, but this may be a heavy operation.

Reply:
the only other option I see is

```
        unsafe {
            std::ptr::drop_in_place(&mut self.db);
            std::ptr::write(&mut self.db, create_new_db());
        }
```

Reply:
Maye we can use the unsafe snippet here but check with Miri first if the code looks ok and sound?
If you want I can have a look at it?